soroban-env-host/src/host/ledger_info_helper.rs

```diff
@@ -16,16 +16,18 @@ impl Host {
                 self.with_ledger_info(|li: &LedgerInfo| Ok(li.min_persistent_entry_ttl))?
             }
         };
-        Ok(ledger_seq.saturating_add(min_live_until.saturating_sub(1)))
+        ledger_seq
+            .checked_add(min_live_until.saturating_sub(1))
+            .ok_or_else(|| self.err_arith_overflow())
```

Comment:
This should probably be an internal error as well, as it means that we probably have a bad misconfiguration with huge min TTL.

Reply:
That wouldn't be "internal" error though. Like, if someone (say in their own test network) sets up a bad min TTL, it's not the host's internal logic bug. The error comes from "external" to the host and arithmetic overflow is the most accurate code to surface it back.

Reply:
Any error caused by misconfiguration is currently an internal error, which makes sense: we don't really want such errors to be recoverable and we do want to catch them as early as possible. Otherwise, the errors represent the conditions caused by the host function user input. I realize that philosophically you may say that network updates are also caused by the 'user' actions, but these are not the same 'user' as the regular Soroban user. In any case, even in your scenario it would be beneficial to return an internal error here - it will quickly identify the misconfiguration instead of hiding it behind a recoverable error. FWIW misconfiguration in this particular case will completely brick the network.

Reply:
Sounds good, updated to `InternalError` and added a comment.